app/models/manageiq/providers/azure/cloud_manager/event_catcher/runner.rb

```diff
@@ -0,0 +1,57 @@
+class ManageIQ::Providers::Azure::CloudManager::EventCatcher::Runner <
+  ManageIQ::Providers::BaseManager::EventCatcher::Runner
+  include ManageIQ::Providers::Azure::EventCatcherMixin
+
+  def event_monitor_handle
+    unless @event_monitor_handle
+      client_id             = @ems.authentication_userid
+      client_key            = @ems.authentication_password
+      tenant_id             = @ems.azure_tenant_id
+      azure_region          = @ems.provider_region
+      @event_monitor_handle = ManageIQ::Providers::Azure::CloudManager::EventCatcher::Stream.new(
+        client_id, client_key, azure_region, tenant_id)
+    end
+    @event_monitor_handle
+  end
+
+  def reset_event_monitor_handle
+    @event_monitor_handle = nil
+  end
+
+  def stop_event_monitor
+    @event_monitor_handle.try(:stop)
+  ensure
+    reset_event_monitor_handle
+  end
+
+  def monitor_events
+    event_monitor_handle.start
+    event_monitor_handle.each_batch do |events|
+      event_monitor_running
+      if events && !events.empty?
```

Comment:
On that note, why would the handle even emit an empty set of events (or nil for that matter)....more a question for @blomquisg if this is a normal pattern